instruments/minghe/mhs5200a.py

```diff
@@ -0,0 +1,226 @@
+        def amplitude(self):
+            """
+            Gets/Sets the amplitude of this channel.
+
+            :units: As specified (if a `~quantities.Quantity`) or assumed to be
+            of units volt.
+            :type: `~quantities.Quantity`
+            """
+            query = ":r{0}a".format(self._chan)
+            response = self._mhs.query(query)
+            return float(response.replace(query, ""))/100.0*pq.V
+
+        @amplitude.setter
+        def amplitude(self, new_val):
+            new_val = 100*assume_units(new_val, pq.V).rescale(pq.V).magnitude
+            query = ":s{0}a{1}".format(self._chan, int(new_val))
+            response = self._mhs.query(query)
+
+        @property
+        def duty_cycle(self):
+            """
+            Gets/Sets the duty cycle of this channel.
+
+            :units: As specified (if a `~quantities.Quantity`) or assumed to be
+            of units seconds.
+            :type: `~quantities.Quantity`
+            """
+            query = ":r{0}d".format(self._chan)
+            response = self._mhs.query(query)
+            duty = float(response.replace(query, ""))*pq.s
+            return duty
+
+        @duty_cycle.setter
+        def duty_cycle(self, new_val):
+            new_val = assume_units(new_val, pq.s).rescale(pq.s).magnitude
+            query = ":s{0}d{1}".format(self._chan, int(new_val))
+            response = self._mhs.query(query)
+
+        @property
+        def enable(self):
+            """
+            Gets/Sets the enable state of this channel.
+
+            :param bool new_val: the enable state
+            """
+            query = ":r{0}b".format(self._chan)
+            return int(self._mhs.query(query).replace(query, "").
+                       replace("\r", ""))
+
+        @enable.setter
+        def enable(self, new_val):
+            query = ":s{0}b{1}".format(self._chan, int(new_val))
+            response = self._mhs.query(query)
+
+        @property
+        def frequency(self):
+            """
+            Gets/Sets the frequency of this channel.
+
+            :units: As specified (if a `~quantities.Quantity`) or assumed to be
+            of units hertz.
+            :type: `~quantities.Quantity`
+            """
+            query = ":r{0}f".format(self._chan)
+            response = self._mhs.query(query)
+            freq = float(response.replace(query, ""))*pq.Hz
+            return freq/100.0
+
+        @frequency.setter
+        def frequency(self, new_val):
+            new_val = assume_units(new_val, pq.Hz).rescale(pq.Hz).\
+                          magnitude*100.0
+            query = ":s{0}f{1}".format(self._chan, int(new_val))
+            response = self._mhs.query(query)
+
+        @property
+        def offset(self):
+            """
+            Gets/Sets the offset of this channel.
+
+            :units: As specified (if a `~quantities.Quantity`) or assumed to be
+            of fraction.
+            :type: `~quantities.Quantity`
```

Comment:
Docstring here says that the offset is a `quantities.Quantity` unitful object, but the getter does not return units, and the setter does not handle unitful values.